### PR TITLE
P4TC - Support const entries

### PR DESCRIPTION
**Pull request by komaljai**

### CODE REVIEW

File: backends/tc/backend.cpp [hunk 1]
```diff
@@ -271,6 +271,53 @@ void ConvertToBackendIR::postorder(const IR::P4Action *action) {
     }
 }
 
+void ConvertToBackendIR::updateConstEntries(const IR::P4Table *t, IR::TCTable *tabledef) {
+    // Check if there are const entries.
+    auto entriesList = t->getEntries();
+    if (entriesList == nullptr) return;
+    auto keys = t->getKey();
+    for (auto e : entriesList->entries) {
+        if (keys == nullptr) {
```
Comment (usha1830) [inline backends/tc/backend.cpp:280]: This if can be moved outside for as it is obtained outside the loop only

File: backends/tc/backend.cpp [hunk 2]
```diff
@@ -271,6 +271,53 @@ void ConvertToBackendIR::postorder(const IR::P4Action *action) {
...
+            return;
+        }
+        auto keyset = e->getKeys();
+        if (keyset->components.size() != keys->keyElements.size()) {
+            ::error("No of keys in const_entries should be same as no of keys in the table.");
```
Comment (usha1830) [inline backends/tc/backend.cpp:285]: Please add error type.

**Review by usha1830 — APPROVED** [review]
LGTM!

File: backends/tc/backend.cpp (hunk 2 above)
Comment (komaljai) [reply backends/tc/backend.cpp:285]: addressed

File: backends/tc/backend.cpp (hunk 1 above)
Comment (komaljai) [reply backends/tc/backend.cpp:280]: addressed

**Pull request merged**